pkg/connector/group.go

```diff
@@ -121,6 +121,37 @@ func (o *groupResourceType) etagMd(group *okta.Group) (*v2.ETagMetadata, error)
 	return nil, nil
 }
 
+func (o *groupResourceType) groupEntitlementMetadata(group *okta.Group) *v2.EntitlementImmutable {
+	if group.Type != "" {
+		if group.Type == "BUILT_IN" {
+			data, err := structpb.NewStruct(map[string]interface{}{
+				"type":               group.Type,
+				"is_system_built_in": true,
+			})
+			if err != nil {
+				return nil
+			}
+			return &v2.EntitlementImmutable{
+				SourceId: group.Id,
+				Metadata: data,
+			}
+		} else {
+			data, err := structpb.NewStruct(map[string]interface{}{
+				"type":               group.Type,
+				"is_system_built_in": false,
+			})
+			if err != nil {
+				return nil
+			}
+			return &v2.EntitlementImmutable{
```

Comment:
we shouldn't include this  if its not actually built in?  presence of the the annotation == its immutable